### PR TITLE
initial stab at new connector for kinesis streams #58

### PR DESCRIPTION
**Pull request by rbraley**
This is the general structure of the Connector for Kinesis data streams, however there are some issues which ought to be resolved. Addresses issue #58 

- proper tests
- handling of generic records
- not leaking underlying library types in the signature of ProducerRecords
- Read side as well.

### CODE REVIEW

**Comment by rbraley** [comment]
Work in progress :)

File: connectors/kinesis-data-streams-connector/src/main/scala/zio/connect/kinesisDataStreams/KinesisDataStreamsConnector.scala [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package zio.connect.kinesisDataStreams
+
+import zio.connect.kinesisDataStreams.KinesisDataStreamsConnector.{KinesisDataStreamsException, ProducerRecord}
+import zio.prelude.Newtype
+import zio.stream.ZSink
+import zio.{Chunk, Trace}
+
+trait KinesisDataStreamsConnector[T] {
+
+  def sinkChunked()(implicit
```
Comment (adrianfilip) [inline connectors/kinesis-data-streams-connector/src/main/scala/zio/connect/kinesisDataStreams/KinesisDataStreamsConnector.scala:10]: Should drop the ()

File: connectors/kinesis-data-streams-connector/src/main/scala/zio/connect/kinesisDataStreams/LiveKinesisDataStreamsConnector.scala [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package zio.connect.kinesisDataStreams
+import izumi.reflect.Tag
+import zio.connect.kinesisDataStreams.KinesisDataStreamsConnector.{KinesisDataStreamsException, ProducerRecord}
+import zio.stream.ZSink
+import zio.{Chunk, Trace, ZIO, ZLayer}
+import nl.vroste.zio.kinesis.client.Producer
+
+final case class LiveKinesisDataStreamsConnector[T](producer: Producer[T]) extends KinesisDataStreamsConnector[T] {
+  override def sinkChunked()(implicit
```
Comment (adrianfilip) [inline connectors/kinesis-data-streams-connector/src/main/scala/zio/connect/kinesisDataStreams/LiveKinesisDataStreamsConnector.scala:9]: Should drop the ()

File: connectors/kinesis-data-streams-connector/src/main/scala/zio/connect/kinesisDataStreams/TestKinesisDataStreamsConnector.scala [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import zio.connect.kinesisDataStreams.TestKinesisDataStreamsConnector.TestKinesisDataStream
+import zio.stm.{STM, TRef, ZSTM}
+import zio.stream.ZSink
+import zio.{Chunk, Trace, ZIO, ZLayer}
+
+private[kinesisDataStreams] final case class TestKinesisDataStreamsConnector[T](
+  kinesisDataStream: TestKinesisDataStream[T]
+) extends KinesisDataStreamsConnector[T] {
+
+  override def sinkChunked()(implicit
```
Comment (adrianfilip) [inline connectors/kinesis-data-streams-connector/src/main/scala/zio/connect/kinesisDataStreams/TestKinesisDataStreamsConnector.scala:18]: () can be removed

File: connectors/kinesis-data-streams-connector/src/main/scala/zio/connect/kinesisDataStreams/KinesisDataStreamsConnector.scala [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package zio.connect.kinesisDataStreams
```
Comment (adrianfilip) [inline connectors/kinesis-data-streams-connector/src/main/scala/zio/connect/kinesisDataStreams/KinesisDataStreamsConnector.scala:1]: I would drop the camelcase format. We should have `zio.connect.kinesisdatastreams` instead.

**Comment by adrianfilip** [comment]
Looks great 🙏 . Only a few minor observations.

File: connectors/kinesis-data-streams-connector/src/main/scala/zio/connect/kinesisDataStreams/package.scala [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package zio.connect
+
+import izumi.reflect.Tag
+import zio.connect.kinesisDataStreams.KinesisDataStreamsConnector.{KinesisDataStreamsException, ProducerRecord}
+import zio.stream.ZSink
+import zio.{Chunk, Trace}
+
+package object kinesisDataStreams {
+  def sinkChunked[T]()(implicit
```
Comment (adrianfilip) [inline connectors/kinesis-data-streams-connector/src/main/scala/zio/connect/kinesisDataStreams/package.scala:9]: () can be removed

File: connectors/kinesis-data-streams-connector/src/main/scala/zio/connect/kinesisdatastreams/KinesisDataStreamsConnector.scala [hunk 6]
```diff
@@ -0,0 +1,27 @@
+package zio.connect.kinesisdatastreams
+
+import zio.connect.kinesisdatastreams.KinesisDataStreamsConnector.{KinesisDataStreamsException, ProducerRecord}
+import zio.prelude.Newtype
+import zio.stream.ZSink
+import zio.{Chunk, Trace}
+
+trait KinesisDataStreamsConnector[T] {
+
+  def sinkChunked(implicit
```
[the remark this answers is not in the document]
Comment (adrianfilip) [reply connectors/kinesis-data-streams-connector/src/main/scala/zio/connect/kinesisdatastreams/KinesisDataStreamsConnector.scala:10]: @svroonland I remember why it is with chunks initially. The api used here produces a sink whose input is in chunks.

[the remark this answers is not in the document]
Comment (adrianfilip) [reply connectors/kinesis-data-streams-connector/src/main/scala/zio/connect/kinesisdatastreams/KinesisDataStreamsConnector.scala:10]: @svroonland  That PR change is not released yet. We will keep an eye on the releases and update it once 0.30.1 is out.

[the remark this answers is not in the document]
Comment (svroonland) [reply connectors/kinesis-data-streams-connector/src/main/scala/zio/connect/kinesisdatastreams/KinesisDataStreamsConnector.scala:10]: Ah yes, I will try to make a release this week and will let you know here.

[the remark this answers is not in the document]
Comment (svroonland) [reply connectors/kinesis-data-streams-connector/src/main/scala/zio/connect/kinesisdatastreams/KinesisDataStreamsConnector.scala:10]: @adrianfilip I just released 0.30.1, should be on Maven soon.

[the remark this answers is not in the document]
Comment (adrianfilip) [reply connectors/kinesis-data-streams-connector/src/main/scala/zio/connect/kinesisdatastreams/KinesisDataStreamsConnector.scala:10]: @svroonland Fantastic. Will upgrade shortly.

**Pull request merged**